.github/workflows/uploadFirrtlReleaseArtifacts.yml

```diff
@@ -36,45 +36,52 @@ jobs:
           files: circt-full-sources.tar.gz*
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+  # This job determines what the build matrix should be.  If this is a
+  # "scheduled" or "workflow dispatch" job, then only build Linux images.  If it
+  # is anything else (i.e., release) then build everything.
+  choose-matrix:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Add Linux
+        id: add-linux
+        run: echo 'out={"os":"linux","runner":"ubuntu-20.04","arch":"x64","tar":"tar czf","archive":"tar.gz","sha256":"shasum -a 256","cont":"\\","setup":"","cmake-args":"-DCMAKE_C_COMPILER=clang-12 -DCMAKE_CXX_COMPILER=clang++-12"}' >> $GITHUB_OUTPUT
+      - name: Add macOS
+        id: add-macos
+        run: echo 'out={"os":"macos","runner":"macos-11","arch":"x64","tar":"gtar czf","archive":"tar.gz","sha256":"shasum -a 256","cont":"\\","setup":"","cmake-args":""}' >> $GITHUB_OUTPUT
+      - name: Add Windows
+        id: add-windows
+        run: echo 'out={"os":"windows","runner":"windows-2019","arch":"x64","tar":"tar czf","archive":"zip","sha256":"sha256sum","cont":"`","setup":"./utils/find-vs.ps1","cmake-args":""}' >> $GITHUB_OUTPUT
+      - name: Build JSON Payloads
+        id: build-json-payloads
+        run: |
+          echo '{"mode":"release","assert":"OFF","shared":"OFF","stats":"ON"}' | jq -sc . > build_configs.json
+          echo build_configs=`cat build_configs.json` >> $GITHUB_OUTPUT
+          echo '${{ steps.add-linux.outputs.out }}' '${{ steps.add-macos.outputs.out }}' '${{ steps.add-windows.outputs.out }}' | jq -sc . > runners.json
+          echo runners=`cat runners.json` >> $GITHUB_OUTPUT
+    outputs:
+      build_configs: ${{ steps.build-json-payloads.outputs.build_configs }}
+      runners: ${{ steps.build-json-payloads.outputs.runners }}
+
+  debug:
+    needs:
+      - choose-matrix
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Print stuff
+        run: |
+          echo "---------------------------------------- Build Config Matrix:"
+          echo '${{ needs.choose-matrix.outputs.build_configs }}' | jq
+          echo "---------------------------------------- Runners Matrix:"
+          echo '${{ needs.choose-matrix.outputs.runners }}' | jq
+
```

Comment:
This is good and useful, it's also worth considering formatting the stuff as markdown and putting it in `GITHUB_STEP_SUMMARY` which allows it to show up pretty in the Actions page, eg.

What it can look like: https://github.com/chipsalliance/chisel/actions/runs/5660492181
How that is done: https://github.com/chipsalliance/chisel/blob/28e235e22bda5e44aaff197baacee1e67b6d86e1/.github/workflows/release-notes.yml#L41

Reply:
Great thinking. I switched to using this. Example output: https://github.com/llvm/circt/actions/runs/5745508781/attempts/1#summary-15573610699